### ISSUE TITLE
[Feature Request] Support System.Threading.Tasks.Parallel

### ISSUE THREAD

**Issue by HydrogenC** [beefytech/Beef#85]
Creating threads manually is not very convenient if I want to go though all the members in a container parallelly. Please implement System.Threading.Tasks.Parallel like C# to make it easy to go through a container in parallel. 
[C# API Reference](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.parallel?view=netframework-4.8)

**Comment by HydrogenC** [comment]
Hi, as I want to attempt to implement this, I wish to know if there is a way to call a Beef function within C++, such as pass a function pointer to C++ as a void* and cast it back in C++? THANKS!

**Comment by bfiete** [comment]
Yes. The 'function' type is just a raw function pointer, you can see an example here where we pass a callback to libcurl: https://github.com/beefytech/Beef/blob/master/IDE/src/util/Transfer.bf#L128

Parallel.For requires a delegate which can't trivially be casted to a C function pointer since a delegate requires state, which COULD be implemented with FFILIB.ClosureAlloc. It's on the "todo" list to be able to directly allocate a a function pointer through any delegate via auto-generated FFILIB.ClosureAlloc calls, but it's not there yet.

Anyway- the actual source to Microsoft's implementation is MIT-licensed here: https://github.com/microsoft/referencesource/blob/master/mscorlib/system/threading/Tasks/Parallel.cs

It's quite a large chunk of work to attempt to take on its conversion, however.
